qendpoint-store/src/main/java/com/the_qa_company/qendpoint/store/experimental/QEPSailStore.java

```diff
@@ -254,5 +257,29 @@ public QEPCloseableIteration<? extends Statement, SailException> getStatements(R
 						return null;
 					});
 		}
+
+		@Override
+		public CloseableIteration<? extends Statement> getStatements(StatementOrder statementOrder, Resource subj,
+				IRI pred, Value obj, Resource... contexts) throws SailException {
+			return getStatements(subj, pred, obj, contexts);
+		}
+
+		@Override
+		public Set<StatementOrder> getSupportedOrders(Resource subj, IRI pred, Value obj, Resource... contexts) {
+//			if (subj == null) {
+			return Set.of(StatementOrder.S);
+//			}
+//			return Set.of();
+		}
+
+		@Override
+		public Comparator<Value> getComparator() {
+			return new Comparator<Value>() {
+				@Override
+				public int compare(Value o1, Value o2) {
+					return o1.toString().compareTo(o2.toString());
+				}
+			};
+		}
```

Comment:
This is the main integration point for the merge join to function.

 - a method to retrieve data sorted in a particular order (since everything is sorted by S and we only support S then it's currently just a pass through)
 - a method to return the possible orders that the underlying data structure can provide for a given statement pattern
 - the comparator that is equivalent to the one used when building the underlying HDT indexes